src/managers/AdManager/AdManager.ts

```diff
@@ -0,0 +1,135 @@
+import { MAX_TIMEOUT_MS, SLOT_NAME, TPL_CODE } from 'consts/dlApi';
+import {
+  COULDNT_FETCH_PRODUCTS_DATA,
+  ERROR_PROMOTED_PRODUCTS_MSG,
+  REQUEST_TIMED_OUT,
+} from 'consts/messages';
+import {
+  extractProductsIds,
+  getProductsIds,
+} from 'managers/AdManager/AdManager.utils';
+import { TPages } from 'types/pages';
+import { TAdProduct, TProductResponse, TFormattedProduct } from 'types/product';
+import getMessage from 'utils/formatters/getMessage';
+import prepareProductsQuery from 'utils/queries/prepareProductsQuery';
+
+class AdManager {
+  private page: TPages | null;
+  private productsIds: number[];
+
+  constructor(page: TPages | null) {
+    this.page = page;
+
+    if (this.page) {
+      this.productsIds = getProductsIds(this.page);
+    }
+  }
+
+  public getPromotedProducts = async () => {
+    if (!dlApi.fetchNativeAd)
+      throw new Error(getMessage(ERROR_PROMOTED_PRODUCTS_MSG));
+
+    let products: TAdProduct[] | null | Error = null;
+
+    const timeoutPromise = new Promise<void>((_, reject) => {
+      setTimeout(() => {
+        reject(getMessage(REQUEST_TIMED_OUT));
+      }, MAX_TIMEOUT_MS);
+    });
+
+    const fetchNativeAd = new Promise<TAdProduct[]>((resolve, reject) => {
+      dlApi.cmd = dlApi.cmd || [];
+      dlApi.cmd.push(async (dlApiObj) => {
+        try {
+          const ads = await dlApiObj.fetchNativeAd!({
```

Comment:
Pobranych produktów może być więcej niż jeden. Zawsze powinniśmy brać pierwszy zwrócony.
Jeśli ktoś w konfiguracji ustawi, że chce wyświetlić więcej niż jeden, to powinnyśmy wywołać funkcję tyle razy ile produktów wyświetlić, a co za tym idzie slotów osadzić. Każde wywołanie osadza osobny slot.
Jeśli będziemy mieć w jednym slocie pięć produktów, to zliczymy jakby user widział jeden, a nie pięć.

Tutaj przykład takiego zachowania:
https://pomoc.onetads.pl/reference/specyfikacje-dla-sklepow#obs%C5%82uga-odpowiedzi